File: 0020-valid-parentheses/0020-valid-parentheses.py
class Solution(object):
    def isValid(self, s):
        a=["(","{","["]
        b=[0]
        for x in s:
            if x in a:
                b.append(x)
            elif x == ")" and b and b[-1] == "(":
                b.pop()
            elif x == "}" and b and b[-1] == "{":
                b.pop()
            elif x == "]" and b and b[-1] == "[":
                b.pop()
            else :
                return False
        if b==[0]:
            return True
        else:
            return False

File: 0020-valid-parentheses/0020-valid-parentheses.cpp
class Solution {
public:
    bool isValid(string s) {
        int len = s.length();
    char a[len+1];
    int index=0;
    for (char x:s)
    {
        if (x=='['||x=='{'||x=='(')
        {
            a[index]=x;
            index++;
        }
        else if (index>0 && x==']' && a[index-1]=='[')
        {
            a[index-1]='0';
            index--;
        }
        else if (index>0 && x=='}' && a[index-1]=='{')
        {
            a[index-1]='0';
            index--;
        }
        else if (index>0 && x==')' && a[index-1]=='(')
        {
            a[index-1]='0';
            index--;
        }
        else{return false;}
    }
    if (index==0)
    {return true;}
    return false;
        
    }
};